lambda_function.py

```diff
@@ -46,7 +46,7 @@ def get_messages_from_json_event(event) -> list[dict]:
     messages = [json.loads(sqs_record["body"]) for sqs_record in sqs_records]
     non_deleted_messages = [message
                             for message in messages
-                            if message["status"] is not "Deleted" and message["tableItemIdentifier"] is not ""]
+                            if message["status"] != "Deleted" and message["tableItemIdentifier"] != ""]
```

Comment:
I was looking at the variable name `non_deleted_messages` which isn't 100% correct and that got me thinking:

Initially when I wrote this, my thinking was "since `Deleted` messages have an empty `tableItemIdentifier`, we should exclude them since we can't do anything with an empty string, but if, in the future, we changed that, then it should proceed"; the logic should have been just `if message["tableItemIdentifier"] != ""]` but for some reason, I wrote what I did above (maybe I changed my mind last minute)

1. My question is, was my initial assumption correct or should we exclude `Deleted` message regardless of whether they have `tableItemIdentifier` or not?

2. If a message doesn't have a `tableItemIdentifier` then it's also useless, but that might be a mistake (a `Created` or `Updated` message should have a `tableItemIdentifier`); should we throw an error in that case or do we just want to silently exclude these? 
3. Should we rename `non_deleted_messages` to `non_deleted_messages_with_table_identifier` or `messages_with_table_identifier` depending on what you decide for question 1 & 2

Reply:
1. I think it's better to be explicit. We definitely don't care about deleted items so it makes sense to explicitly exclude them.
2. I think while everything is settling it'll be useful to know if we've got a missing `tableItemIdentifier` for a Create or Update message. I've stopped filtering out the empty ones and changed it to raise an error if the tableItemidentifier is empty. It will already raise an erro r if it's missing.
3. I think non_deleted_messages is now fine as we're no longer filtering out empty tableItemIdentifiers.